### PR TITLE
Validator registration filter non active keys before submission

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Bug fix

**What does this PR do? Why is it needed?**

some relays do not accept validators that are not in active status such as validators that are exited. 
currently, we send validator registrations as long as the public key has an index but it seems that some relays return 502 if the validator has exited. this pr adds additional filtering for these cases. 

currently the user will receive this kind of error
```
[2023-04-25 12:08:38]  WARN validator: Push proposer settings error error=could not submit signed registrations to beacon node: rpc error: code = InvalidArgument desc = Could not register block builder: could not register validator(s): not a known validator: 0x8b91f34c039c348f73c2dda492998e268c2815a433f59a6667258267739947dcfc8ade6823b3f1f3cfef824497eb113c: recv 400 BadRequest response from API: did not receive 200 response from API: Builder API validator registration unsuccessful
```

now the non active status validators should be filtered and not included when submitting builder validator registration
```
[2023-04-25 14:26:43]  INFO validator: Validator exited index=392139 pubKey=0x8b91f34c039c status=EXITED
[2023-04-25 14:26:43]  INFO validator: Validator exited index=392137 pubKey=0x855ae9c6184d status=EXITED
[2023-04-25 14:26:43]  INFO validator: Validator activated index=392138 publicKey=0xaa0ef7404c3a
[2023-04-25 14:26:43]  INFO validator: Attestation schedule attesterDutiesAtSlot=1 pubKeys=[0xaa0ef7404c3a] slot=5495244 slotInEpoch=12 timeTillDuty=2m5s totalAttestersInEpoch=1
[2023-04-25 14:26:43]  INFO validator: Validator client started with provided proposer settings that sets options such as fee recipient and will periodically update the beacon node and custom builder (if --enable-builder)
[2023-04-25 14:26:43]  INFO validator: Submitted builder validator registration settings for custom builders
```
the most expensive call added here is `MultipleValidatorStatus` , below is a screenshot of the api called in another place in the code with 1000 validators on prater resulting in 20millisecond return
image

### CODE REVIEW

File: validator/client/validator.go [hunk 1]
```diff
@@ -1128,6 +1150,12 @@ func (v *validator) buildSignedRegReqs(ctx context.Context, pubkeys [][fieldpara
 			continue
 		}
 
+		// map is populated before this function in buildPrepProposerReq
+		_, ok := v.pubkeyToValidatorIndex[k]
+		if !ok {
+			continue
+		}
```
Comment (prestonvanloon) [inline validator/client/validator.go:1157]: Do you need a read lock to access this map?

Comment (james-prysm) [reply validator/client/validator.go:1157]: not currently, is this something that should have a read lock and write lock for updating?

Comment (james-prysm) [reply validator/client/validator.go:1157]: access to this map should be sequential only so read and writes shouldn't overlap in this case I think we won't need the lock but will test.

File: validator/client/validator.go [hunk 2]
```diff
@@ -1030,36 +1034,50 @@ func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKey
...
 			}
-
 			if !ok { // Nothing we can do if RPC server doesn't have validator index.
 				continue
 			}
-
-			validatorIndex = i
 			v.pubkeyToValidatorIndex[k] = i
 		}
+		status, err := v.validatorClient.ValidatorStatus(ctx, &ethpb.ValidatorStatusRequest{PublicKey: k[:]})
```
Comment (nisdas) [inline validator/client/validator.go:1052]: doing this for every validator seems expensive, each request will be a separate roundtrip to the beacon node.

Comment (james-prysm) [reply validator/client/validator.go]: I've updated using our internal API which will minimize the requests.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**